src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -3574,6 +3596,12 @@ extern "C" void * QCALLTYPE CallCatchFunclet(QCall::ObjectHandleOnStack exceptio
 #ifdef HOST_WINDOWS
         if ((pLongJmpBuf == NULL) && !IsComPlusException(&lastExceptionRecord) && MapWin32FaultToCOMPlusException(&lastExceptionRecord) == kSEHException)
         {
+#if defined(HOST_X86)
```

Comment:
Note this implementation will cause problems in specific case when handling foreign exceptions - issue that I've fixed in #113323. But I don't see a way to address that on x86 windows.

Reply:
Windows X86 has no mechanism to unwind registers other than what is in the SEH record itself on the stack (pointer to registeration routine and the stack pointer itself). Hence the C++ exception handling doesn't depend on it either. I did actually implement the logic to handle #113323. We don't have to trick the unwinder with a special frame since there's no unwinder, just the SEH chain on the stack pointed by `fs:[0]`. Instead we pop all the SEH frames up to the context SP and then re-raise the exception from current frame.

Reply:
Ah, that's right, I haven't realized that popping those will result in the effect we need.

Reply:
Took me longer than I am willing to admit to come up with that solution 😅